packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -326,15 +329,28 @@ export async function getPageStaticInfo(params: {
 
     const requiresServerRuntime = ssr || ssg || pageType === 'app'
 
-    resolvedRuntime =
-      SERVER_RUNTIME.edge === resolvedRuntime
-        ? SERVER_RUNTIME.edge
-        : requiresServerRuntime
-        ? resolvedRuntime || nextConfig.experimental?.runtime
-        : undefined
+    resolvedRuntime = isEdgeRuntime(resolvedRuntime)
+      ? resolvedRuntime
+      : requiresServerRuntime
+      ? resolvedRuntime || nextConfig.experimental?.runtime
+      : undefined
+
+    if (resolvedRuntime === SERVER_RUNTIME.experimentalEdge) {
+      warnAboutExperimentalEdge()
+    }
 
-    if (resolvedRuntime === SERVER_RUNTIME.edge) {
-      warnAboutExperimentalEdgeApiFunctions()
+    if (
+      resolvedRuntime === SERVER_RUNTIME.edge &&
+      pageType === 'pages' &&
+      page &&
+      !isAPIRoute(page)
+    ) {
+      Log.error(
+        `Page ${page} provided runtime 'edge', the edge runtime for rendering is currently experimental. Use runtime 'experimental-edge' instead.`
+      )
+      if (!isDev) {
```

Comment:
Should this throw an error in development in order for it to not continue to rendering?

Reply:
As discussed online, it's actually not easy because the page hangs when throwing an exception here, since this code is used during webpack compilation.